protocols/hawakeningmaster.js

```diff
@@ -0,0 +1,718 @@
+import Core from './core.js'
+// import Ajv from 'ajv'
+// const ajv = new Ajv()
+
+/**
+ * Implements the protocol for retrieving a master list for Hawakening, a fan project of the UnrealEngine3 based game HAWKEN
+ * using a Meteor backend for the master server
+ */
+export default class hawakeningmaster extends Core {
+  constructor () {
+    super()
+
+    // backend API url for original Hawken release
+    // const meteorUri = 'https://v2-services-live-pc.playhawken.com'
+    // Hawakening API for public release in 2024
+    const meteorUri = 'https://hawakening.com/api'
+
+    this.backendApi = new MeteorBackendApi(this, meteorUri)
+    this.backendApi.setLogger(this.logger)
+
+    // set when querying needs access token
+    this.requireToken = false
+    // set when querying for specific server only
+    this.doQuerySingle = false
+    // set to logout on cleanup (to revoke access token)
+    this.doLogout = true
+
+    // stored user, queried from backend
+    this.userInfo = null
+
+    // Don't use the tcp ping probing
+    this.usedTcp = true
+  }
+
+  async run (state) {
+    await this.retrieveClientAccessToken()
+    await this.retrieveUser()
+
+    await this.queryInfo(state)
+    await this.cleanup(state)
+  }
+
+  async queryInfo (state) {
+    if (this.doQuerySingle) {
```

Comment:
For now, please move this into the game protocol file.

Reply:
What do you mean?

Reply:
I meant to move the `doQuerySingle` logic from master file to game protocol file, but on a second thought this looks good.